src/QLContext.actor.cpp

```diff
@@ -701,14 +701,15 @@ Optional<IndexInfo> UnboundCollectionContext::getSimpleIndex(std::string simpleI
 	}
 }
 
-Optional<IndexInfo> UnboundCollectionContext::getCompoundIndex(IndexInfo prefix, std::string nextIndexKey) {
+Optional<IndexInfo> UnboundCollectionContext::getCompoundIndex(std::vector<std::string> const& prefix,
+                                                               std::string nextIndexKey) {
 	if (bannedFieldNames.find(nextIndexKey) != bannedFieldNames.end())
 		return Optional<IndexInfo>();
-	auto indexV = simpleIndexMap.find(prefix.indexKeys[0].first);
+	auto indexV = simpleIndexMap.find(prefix[0]);
 	ASSERT(indexV != simpleIndexMap.end());
 	for (IndexInfo index : indexV->second) {
-		if (index.size() > prefix.size() && index.hasPrefix(prefix)) {
-			if (index.indexKeys[prefix.indexKeys.size()].first == nextIndexKey) {
+		if (index.hasPrefix(prefix)) {
+			if (index.indexKeys[prefix.size()].first == nextIndexKey) {
```

Comment:
This is a potential out of bound exception, when `prefix.size()==indexKeys.size()`. In that case, `hasPrefix()` could still return `true` and boom we have a problem here then. I think we still need the explicit size compare.

Reply:
That check still exists, its moved to `hasPrefix()`.

Reply:
Gotcha! will fix it.

Reply:
The check in `hasPrefix()` is necessary for the purpose of that method, i.e. size of the indexKeys needs to be at lease same as the size of the prefix, but here we need the size of indexKeys to be at lease larger than the size of prefix by at least 1, otherwise it will be out of bound.